avocado/utils/asset.py

```diff
@@ -71,21 +71,178 @@ def __init__(self, name, asset_hash, algorithm, locations, cache_dirs,
         :param expire: time in seconds for the asset to expire
         :param metadata: metadata which will be saved inside metadata file
         """
-        self.name = name
+        parsed_url = urlparse.urlparse(name)
+        # Asset file name
+        self.basename = os.path.basename(parsed_url.path)
```

Comment:
Could rename `self.basename` to `self.asset_name`?

`basename` seems to vague to me.

Reply:
I kept it from the original class, but I don't have any problem with changing it. Note taken for V3.